std/http/headers.zig

```diff
@@ -0,0 +1,603 @@
+// HTTP Header data structure/type
+// Based on lua-http's http.header module
+//
+// Design criteria:
+//   - the same header field is allowed more than once
+//       - must be able to fetch separate occurrences (important for some headers e.g. Set-Cookie)
+//       - optionally available as comma separated list
+//   - http2 adds flag to headers that they should never be indexed
+//   - header order should be recoverable
+//
+// Headers are implemented as an array of entries.
+// An index of field name => array indices is kept.
+
+const std = @import("../std.zig");
+const debug = std.debug;
+const assert = debug.assert;
+const testing = std.testing;
+const mem = std.mem;
+const Allocator = mem.Allocator;
+
+fn never_index_default(name: []const u8) bool {
+    if (mem.eql(u8, "authorization", name)) return true;
+    if (mem.eql(u8, "proxy-authorization", name)) return true;
+    if (mem.eql(u8, "cookie", name)) return true;
+    if (mem.eql(u8, "set-cookie", name)) return true;
```

Comment:
Headers are usually Title-Case, e.g. `Set-Cookie`.

Reply:
In HTTP/2, all headers "MUST" be lower case (see https://httpwg.org/specs/rfc7540.html#HttpHeaders).
Additionally title casing can be quite expensive computationally.

Modern HTTP libraries use lower case by default to both match HTTP/2, and save CPU cycles.